### PR TITLE
Tpm2 option revamp

### PR DESCRIPTION
**Pull request by williamcroberts**

### CODE REVIEW

**Comment by williamcroberts** [comment]
@martinezjavier Here it is, the PR of doom. On the plus side, we removed 800 lines of code, I love that.

**Comment by martinezjavier** [comment]
@williamcroberts great, negative diffstats are the best :smiley:

File: lib/tpm2_options.c [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+
+    tpm2_option_code rc = tpm2_option_code_err;
+    bool result = false;
+
+    UNUSED(envp);
+
+    struct option long_options [] = {
+        { "tcti",    required_argument, NULL, 'T' },
+        { "help",    no_argument,       NULL, 'h' },
+        { "verbose", no_argument,       NULL, 'v' },
```
Comment (liuqun) [inline lib/tpm2_options.c]: Bug #967 Common option "-v" should print version, not verbose

File: lib/tpm2_options.c [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+        { "quiet",   no_argument,       NULL, 'Q' },
+        { "version", no_argument,       NULL, 'V' },
```
Comment (liuqun) [inline lib/tpm2_options.c]: "version" should use lower case short option "-v"

**Pull request merged**